starttls-policy/starttls_policy/tests/configure_test.py

```diff
@@ -44,7 +48,41 @@ def test_manual_instructions(self):
                 "instruct_string")
 
 
+class TempPolicyDir(object):
+    # pylint: disable=useless-object-inheritance,attribute-defined-outside-init
+    """This context manager creates temporary directory
+    and writes given policy into it."""
+    def __init__(self, conf):
+        self._conf = conf
+
+    def __enter__(self):
+        self._tmpdir = tempfile.mkdtemp()
+        self._policy_filename = os.path.join(self._tmpdir, 'policy.json')
+        with open(self._policy_filename, 'w') as pol_file:
+            pol_file.write(self._conf)
+        return self._tmpdir
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        os.remove(self._policy_filename)
+        os.rmdir(self._tmpdir)
+
```

Comment:
`mock` is a great tool, but I doubt about it's purpose here. Mocking actual file IO allows us to run tests clearly, without interacting with real FS. But creation of temporary directory with file is not something we can't afford.

In general, best practice is to mock as less as possible, resort to mock only when counterpart our code tested against cannot be efficiently represented by real instance.

Mock solution has following benefit over real FS write: it avoids usage of real FS. But it has drawbacks:
* Reader will **always** read `read_data` with same type of string `mock_open` was initialized. Consider [following examples](https://gist.github.com/Snawoot/33b797bad8265a78f1071626d41de559). So, `mock_open` is imperfect.
* Reading of file by a program may produce different results depending on method of reading. For example, there are no reasons why `policy.json` can't have byte order mark in the beginning of file and to handle such case we have to use special BOM-aware codec. Mocked file defined by a string doesn't allow us to conduct tests against real files.
* Mocking requires test to have some knowledge of internal structure of program in order it can be properly patched, and this knowledge has to be maintained. What if we will use some faster json module which will load json file directly?

BTW, when our project will become popular (I believe!) and policy list will be big enough to not fit in memory, we probably will have to reconsider parsing of json file anyway in order to reduce memory footprint required to generate policy. Maybe generator-like json parser like `ijson` will do.

I think if we will need to mock file IO, we can return to it in future. But for now it is unnecessary.